src/commenter.js

```diff
@@ -122,48 +143,70 @@ export default class Commenter {
   /**
    * Post a comment to the given pull request, and deduplicate (best-effort) comments by purpose.
    *
-   * @param {string} repository The repository on which to create the comment, in slug form.
-   * @param {number|string} pullRequest The pull request number on GitHub which holds the thread
+   * @param {string} owner The repository owner.
+   * @param {string} repo The repository name on which to create the comment.
+   * @param {number|string} issue_number The pull request number on GitHub which holds the thread
    *   where the comment should be created.
    * @param {string|Buffer|Readable} content The GitHub flavored markdown to put in the comment.
+   * @param {?boolean|string} replace Whether to replace the existing comment (true) or just edit it
+   *   in-place (false). If
    * @param {string} purpose The purpose of the comment.
    * @return {Promise<Object>} Some simple metadata about the action performed, suitable for
    *   displaying a simple message detailing the action taken.
    */
-  async postComment({ repository, pullRequest, content, purpose = null }) {
-    const commentToken = purpose && getUniqueToken({ repository, pullRequest, purpose });
+  async postComment({ owner, repo, issue_number, content, replace = false, purpose = null }) {
+    if (replace && !purpose) {
+      throw new Error('replace option enabled but no purpose specified');
+    }
+
+    const commentToken =
+      purpose && getUniqueToken({ repository: `${owner}/${repo}`, issue_number, purpose });
 
     const stringContent = utils.toString(content);
 
     const existingComment =
       commentToken &&
       (await this.findComment({
-        repository,
-        pullRequest,
+        owner,
+        repo,
+        issue_number,
         commentToken,
       }));
 
     const taggedContent = await stringContent.then((content) =>
       commentToken ? `${getDedupeComment(commentToken)}\n${content}` : content
     );
+
+    const create = () =>
+      this.createComment(
+        {
+          owner,
+          repo,
+          issue_number,
+        },
+        taggedContent
+      );
+
     if (existingComment) {
-      if (existingComment.body === taggedContent) {
+      if (replace !== 'force' && existingComment.body === taggedContent) {
         return {
           action: null,
           link: existingComment.html_url,
           user: existingComment.user.login,
         };
       }
 
+      if (replace) {
+        const [, createInfo] = await Promise.all([this.removeComment(existingComment), create()]);
```

Comment:
Does the order in which these are executed not matter because they're just distinct comments from Github's persective?

Reply:
Correct